plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -748,8 +748,21 @@ bool CppParser::parseByJson(
 
   //--- Read the compilation commands compile database ---//
 
-  std::vector<clang::tooling::CompileCommand> compileCommands =
+  std::vector<clang::tooling::CompileCommand> tempCompileCommands =
     compDb->getAllCompileCommands();
+  std::vector<clang::tooling::CompileCommand> compileCommands;
+
+  const std::vector<std::string> cppExts{
+    ".c", ".cc", ".cpp", ".cxx", ".o", ".so", ".a"};
+
+  std::copy_if(tempCompileCommands.begin(), tempCompileCommands.end(),
```

Comment:
We could remove elements with `remove_if` or something similar in order to avoid a copy of the compile commands.